repo/content/content_manager.go

```diff
@@ -945,6 +945,9 @@ type ManagerOptions struct {
 	TimeNow                func() time.Time // Time provider
 	DisableInternalLog     bool
 	PermissiveCacheLoading bool
+	// Whether or not writes, for index compaction and cleanup, are permitted
+	// while reading the indexes.
+	AllowCleanupWritesOnIndexLoad bool
```

Comment:
This option is so the behavior can be specified from the callers, and eventually wired in the CLI.